eclair-core/src/main/scala/fr/acinq/eclair/db/migration/CompareDb.scala

```diff
@@ -0,0 +1,80 @@
+package fr.acinq.eclair.db.migration
+
+import fr.acinq.eclair.db.DualDatabases
+import fr.acinq.eclair.db.pg.PgUtils
+import grizzled.slf4j.Logging
+import scodec.bits.ByteVector
+
+import java.sql.{Connection, ResultSet}
+
+object CompareDb extends Logging {
+
+  def compareTable(conn1: Connection,
+                   conn2: Connection,
+                   table1: String,
+                   table2: String,
+                   hash1: ResultSet => ByteVector,
+                   hash2: ResultSet => ByteVector): Boolean = {
+    val rs1 = conn1.prepareStatement(s"SELECT * FROM $table1").executeQuery()
+    val rs2 = conn2.prepareStatement(s"SELECT * FROM $table2").executeQuery()
+
+    var hashes1 = List.empty[ByteVector]
+    while (rs1.next()) {
+      hashes1 = hash1(rs1) +: hashes1
+    }
+
+    var hashes2 = List.empty[ByteVector]
+    while (rs2.next()) {
+      hashes2 = hash2(rs2) +: hashes2
+    }
+
+    val res = hashes1.sorted == hashes2.sorted
+    val diff1 = hashes1 diff hashes2
+    val diff2 = hashes2 diff hashes1
+
+    if (res) {
+      logger.info(s"tables $table1/$table2 are identical")
+    } else {
+      logger.warn(s"tables $table1/$table2 are different diff1=${diff1.take(3).map(_.toHex.take(128))} diff2=${diff2.take(3).map(_.toHex.take(128))}")
+    }
+
+    res
+  }
+
+  // @formatter:off
+  import fr.acinq.eclair.db.jdbc.JdbcUtils.ExtendedResultSet._
+  def bytes(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector(columnName)
+  def bytesnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorNullable(columnName).getOrElse(ByteVector.fromValidHex("deadbeef"))
+  def hex(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorFromHex(columnName)
+  def hexnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector32FromHexNullable(columnName).map(_.bytes).getOrElse(ByteVector.fromValidHex("deadbeef"))
```

Comment:
Why is this one using `ByteVector32` but the one above uses `ByteVector`? Also its fallback isn't a `ByteVector32`

Reply:
Fixed with 5c3eac546.